remo/reports/api/views.py

```diff
@@ -19,3 +30,88 @@ def retrieve(self, request, pk):
         serializer = ActivitiesDetailedSerializer(report,
                                                   context={'request': request})
         return Response(serializer.data)
+
+
+class ActivitiesKPIFilter(django_filters.FilterSet):
+    """Filter for activities KPI endpoint."""
+    category = django_filters.CharFilter(name='functional_areas__name')
+    initiative = django_filters.CharFilter(name='campaign__name')
+
+    class Meta:
+        model = NGReport
+        fields = ['country', 'category', 'initiative']
+
+
+class ActivitiesKPIView(APIView):
+
+    def get(self, request):
+        """Returns serialized data for Activities KPI"""
+
+        qs = NGReport.objects.filter(report_date__lte=now())
+        activities = ActivitiesKPIFilter(request.query_params, queryset=qs)
+        weeks = int(request.query_params.get('weeks', KPI_WEEKS))
+
+        # Total number of activities to day
+        total = activities.qs.count()
+
+        # Quarter calculations
+        current_quarter_start = get_quarter()[1]
+
+        # Total number of activities for current quarter
+        quarter_total = activities.qs.filter(
+            report_date__gte=current_quarter_start).count()
+
+        # Total number of activities until start of quarter
+        quarter_start_total = activities.qs.filter(
+            report_date__lte=current_quarter_start).count()
+
+        try:
+            # Percentage change of activities since start of quarter
+            diff = total - quarter_start_total
+            percent_quarter = diff/float(quarter_start_total)
+        except ZeroDivisionError:
+            percent_quarter = 0
+
+        # Week calculations
+        current_week_start = now() - timedelta(days=now().weekday())
+        prev_week_start = current_week_start - timedelta(weeks=1)
+
+        # Total number of activities this week
+        week_total = activities.qs.filter(
+            report_date__gte=current_week_start).count()
+        query_range = [prev_week_start, current_week_start]
+
+        # Total number of activities for previous week
+        prev_week_total = activities.qs.filter(
+            report_date__range=query_range).count()
+
+        try:
+            # Percentage change of activities compared with previous week
+            diff = week_total-prev_week_total
+            percent_week = diff/float(prev_week_total)
+        except ZeroDivisionError:
+            percent_week = 0
+
+        weekly_count = []
+        for i in range(weeks):
+            start = current_week_start - timedelta(weeks=i)
+            end = start + timedelta(weeks=1)
+
+            # Total number of activities for previous weeks
+            count = activities.qs.filter(
+                report_date__range=[start, end]).count()
+            weekly_count.append({'week': weeks-i, 'activities': count})
+
+        kwargs = {
```

Comment:
What about renaming a few of the kwargs?
Eg:
quarter_total
quarter_growth_percentage
week_total
week_growth_percentage